File: nim/src/minsk/syntaxToken.nim
import syntaxKind
import syntaxNode
import minskObject

type
  SyntaxToken* = ref object of SyntaxNode
    pKind: SyntaxKind
    position*: int
    text*: string
    value*: MinskObject

proc newToken*(
  kind: SyntaxKind,
  position: int,
  text: string,
  value: MinskObject
): SyntaxToken =
  new(result)
  result.pKind = kind
  result.position = position
  result.text = text
  result.value = value

method kind*(self: SyntaxToken): SyntaxKind =
  self.pKind

method children*(self: SyntaxToken): seq[SyntaxNode] =
  @[]

File: hs/src/Minsk/SyntaxToken.hs
module Minsk.SyntaxToken (SyntaxToken (..)) where

import BasicPrelude
import Minsk.Object (Object)
import Minsk.SyntaxKind (SyntaxKind)

data SyntaxToken = SyntaxToken
  { kind :: SyntaxKind,
    position :: Int,
    text :: Text,
    value :: Maybe Object
  }
  deriving (Show)
